components/setting_item_max.tsx

```diff
@@ -135,27 +57,46 @@ export default class SettingItemMax extends React.PureComponent {
         document.removeEventListener('keydown', this.onKeyDown);
     }
 
-    onKeyDown = (e) => {
-        if (this.props.shiftEnter && e.keyCode === Constants.KeyCodes.ENTER && e.shiftKey) {
+    onKeyDown = (e: KeyboardEvent) => {
+        if (this.props.shiftEnter && e.key === Constants.KeyCodes.ENTER[0] && e.shiftKey) {
             return;
         }
-        if (this.props.disableEnterSubmit !== true && isKeyPressed(e, Constants.KeyCodes.ENTER) && this.props.submit && e.target.tagName !== 'SELECT' && e.target.parentElement && e.target.parentElement.className !== 'react-select__input' && !e.target.classList.contains('btn-cancel') && this.settingList.current && this.settingList.current.contains(e.target)) {
-            this.handleSubmit(e);
+        if (this.props.disableEnterSubmit !== true &&
+            isKeyPressed(e, Constants.KeyCodes.ENTER) &&
+            this.props.submit &&
+            e.target
+        ) {
+            const target = e.target as HTMLElement;
+            if (target.tagName !== 'SELECT' &&
+                target.parentElement &&
+                target.parentElement.className !== 'react-select__input' &&
+                !target.classList.contains('btn-cancel') &&
+                this.settingList.current &&
+                this.settingList.current.contains(target)
+            ) {
+                this.handleSubmit(e as any);
+            }
         }
     }
 
-    handleSubmit = (e) => {
+    handleSubmit = (e: React.MouseEvent<HTMLButtonElement>) => {
         e.preventDefault();
 
-        if (this.props.setting) {
+        if (!this.props.submit) {
+            return;
+        }
+
+        if (this.props.submitWithSettings) {
+            this.props.submitWithSettings([this.props.setting!]);
+        } else if (this.props.setting) {
```

Comment:
I don't really get why we have this extra submit handler here that accepts an array of settings.
Could you elaborate why that is here?